### ISSUE TITLE
Compass/bearing snapping fails when also zooming

### ISSUE THREAD

**Issue by friedbunny** [mapbox/mapbox-gl-native#1064]
If one rotates and then pinch zooms, the pinch gesture [cancels the map transition](https://github.com/mapbox/mapbox-gl-native/blob/9dd9bd6e9ddf40ae3ae1746b191ce7d0a8ea0842/platform/ios/MGLMapView.mm#L644) called by the fancy new snapping (#1059) but `bearing` still gets reset and becomes out of sync. :disappointed:

```
2015-03-22 21:19:55.291 Mapbox GL[5984:764115] Rotate: -2.208035º
2015-03-22 21:19:55.292 Mapbox GL[5984:764115] Should we snap?: 357.791965º
2015-03-22 21:19:55.292 Mapbox GL[5984:764115] SNAP!
2015-03-22 21:19:55.294 Mapbox GL[5984:764115] Pinch: -0.000000º
...
2015-03-22 21:26:44.939 Mapbox GL[5984:764115] Rotate: -3.104398º
2015-03-22 21:26:44.939 Mapbox GL[5984:764115] Should we snap?: 356.895602º
2015-03-22 21:26:44.940 Mapbox GL[5984:764115] SNAP!
2015-03-22 21:26:44.940 Mapbox GL[5984:764115] Pinch: -0.000000º
```
[![screen-shot-2015-03-22-at-9 20 17-pm](https://cloud.githubusercontent.com/assets/1198851/6774503/c3165644-d0de-11e4-92a1-8b563742219a.png)](https://cloud.githubusercontent.com/assets/1198851/6774501/b763a54a-d0de-11e4-8e23-8504be4ffce7.png)

**Comment by incanus** [comment]
Ok, I'll probably end up reverting the feature in `master` then if it's got complications. Can hit it after the first beta...

**Comment by friedbunny** [comment]
Yeah, unfortunately it's not quite there yet. I'll look into putting a `isSnapping` check or something in there.

**Comment by friedbunny** [comment]
I've got a fix ready to go for this (e94c7a305de7e0df6e02ee58089d1ac0e9f50a29) that essentially duplicates `animatingGesture` stuff elsewhere. Not sure that `userInteractionEnabled` needs to be disabled (didn't notice any difference), but I've left that there. Wait on the PR?
